pkg/threatest/parser/main.go

```diff
@@ -0,0 +1,86 @@
+package parser
+
+import (
+	"fmt"
+	"github.com/datadog/threatest/pkg/threatest"
+	"github.com/datadog/threatest/pkg/threatest/detonators"
+	"github.com/datadog/threatest/pkg/threatest/matchers/datadog"
+	"sigs.k8s.io/yaml" // we use this library as it provides a handy "YAMLToJSON" function
+	"strings"
+	"time"
+)
+
+// Parse turns a YAML input string into a list of Threatest scenarios
+// TODO: A SSH configuration shouldn't be required at this point
```

Comment:
I agree it feels a bit cumbersome. Maybe worth elaborating a struct storing this kind of data (maybe with a builder constructor pattern) and exposing a `Parse`  method instead?

Reply:
definitely, the catch is that different attack techniques can only be detonated by some detonators.

e.g. running a command can only be detonated by an SSH or local detonator, while a Stratus Red Team attack technique can only be detonated by the Stratus Red Team detonator.

So it likely requires major changes in the Threatest Go API which are out of scope of this PR, but that we'll need to make better for sure